Xext/panoramiX.c

```diff
@@ -1064,28 +1064,31 @@ ProcXineramaQueryScreens(ClientPtr client)
         swapl(&rep.length);
         swapl(&rep.number);
     }
-    WriteToClient(client, sizeof(xXineramaQueryScreensReply), &rep);
+
+    xXineramaScreenInfo *scratch = calloc(number, sizeof(xXineramaScreenInfo));
+    if (number && !scratch)
```

Comment:
Maybe move the allocation inside the `if (!noPanoramiXExtension) {` block so that a) you don't have to check `number` (duplicate check essentially), and b) you don't attempt to allocate zero bytes (will generally work, but is pointless in this case).

Reply:
good point.